Lib/uuid.py

```diff
@@ -656,7 +656,7 @@ def _random_getnode():
 
 _node = None
 
-def getnode():
+def getnode(*, getters=None):
```

Comment:
I added the `getters` keyword to make testing, but in principle it could be used to extend `getnode` and `uuid1` for some other platform. `uuid1(node=getnode(getters=[my_ifconfig_parser]))`

Reply:
I think the change is fine in principle, however this *is* an API change, so it needs to be documented in the `getnode()` docstring, and in the [uuid.getnode()](https://docs.python.org/3/library/uuid.html#uuid.getnode) documentation.  Don't forget a `versionchanged` in the latter.   Both docs need to also mention that random UUID generation is always included in the list of getters.  Are you okay with not providing a way to override that?  (I think I am.)

The API change also needs to be documented in your News file entry.

Changing the API will also make this more difficult to backport, since you can't change the API in any previous version of Python.  Are you okay with that (i.e. either not backporting it, or implementing the backport differently)?

Reply:
I wasn't sure about whether this counts as an API change or whether this needs to be backported, etc. To sidestep the problems that introduces I removed the API change and changed the approach a bit.